### PR TITLE
feat: onboarding carousel variant testing

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Implements the experiment for onboarding carousel variants as described in [notion](https://www.notion.so/valora-inc/651e35300967405a969d31808247ad07?v=dd3840385dc04284a6e1878b7ae66ccb&p=46ff60a5ad3a437cab5a168ae30a03f6&pm=s). Since at the time of initial app start Firebase may not be available random orders of three cards were implemented using the [Fisher–Yates / Durstenfeld's Shuffle](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle); the 'old' vs 'new' variant were done using Math.random resulting in a total of 12 different variants: 6 old - 6 new.

#### Analytics Implemented - Whole Flow
- onboarding_education_start: used as a sanity check that all variants are displayed equally.
- onboarding_education_complete: used to determine which ordering and variant is the most successful.

#### Analytics Implemented - Per Screen
- onboarding_education_step_impression: identify what value prop was shown on each screen individually.

### Other changes

N/A

### Tested

Tested Locally on iOS

### How others should test

1. With a fresh install - view the carousel items
2. Close the app and reload.
3. Observe carousel items and ensure that order changes (It's random so it's possible items might not change every time).

### Related issues

- RET-382

### Backwards compatibility

Yes

### CODE REVIEW

File: src/account/Education.tsx [hunk 1]
```diff
@@ -42,6 +42,8 @@ interface EducationStep {
   // If set to true, title is displayed at the top
   isTopTitle?: boolean
   text?: string
+  valueProp?: string
```
Comment (jeanregisser) [inline src/account/Education.tsx:45]: nit: `prop` is too much linked to React component properties
```suggestion
  valueProposition?: string
```

File: src/analytics/Events.tsx [hunk 2]
```diff
@@ -66,8 +66,11 @@ export enum SettingsEvents {
 
 export enum OnboardingEvents {
   onboarding_education_start = 'onboarding_education_start',
+  onboarding_education_start_experiment = 'onboarding_education_start_experiment',
   onboarding_education_scroll = 'onboarding_education_scroll',
+  onboarding_education_screen_impression = 'onboarding_education_screen_impression',
```
Comment (jeanregisser) [inline src/analytics/Events.tsx:71]: nit: maybe?
```suggestion
  onboarding_education_step_impression = 'onboarding_education_step_impression',
```

Also does this fire for the initial step?

**Review by jeanregisser — COMMENTED** [review]
Looking great! 👍 

I'm wondering if we should update the `Education` component tests to cover this.

File: src/account/Education.tsx (hunk 1 above)
Comment (MuckT) [reply src/account/Education.tsx]: Fixed in 00620f7aa24e884eef465e40f4eeb61559b5ba95!

File: src/analytics/Events.tsx (hunk 2 above)
Comment (MuckT) [reply src/analytics/Events.tsx]: Fixed in 0465ae5f7b225fba739040f340e0d6909b695709!

**Review by jeanregisser — APPROVED** [review]

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the above task on Latest iOS Test Flight Release build V 1.40.0 and updated status in Test Rails

Observation : 
- The carousel orders changes when user kills and relaunch the app every time and sometimes the order does not change

**Test Rails:**
**iOS 14:** https://valoraapp.testrail.io/index.php?/tests/view/69522

**Device :** iPhone 12(14.7.1)

Thanks.!

**Comment by MuckT** [comment]
> The carousel orders changes when user kills and relaunch the app every time and sometimes the order does not change. 

@ValoraQA this is sounds like it's working as intended. It's possible to get the same order of screens about 1 in 12 times. It also can anecdotally feel like some sets are more common, but over a longer series of displays they should be random. There's a test for this in https://github.com/valora-inc/wallet/pull/2827/files#diff-1d54205b4bf6765a1190603d33ca14497011165f6c100e30b916b18432829e2b.

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the above task on Latest Android Internal Release build V 1.40.0 and updated status in Test Rails

**Test Rails:**
**Android 11:** https://valoraapp.testrail.io/index.php?/tests/view/69553

**Devices :** Google Pixel 2XL(11.0)

Thanks.!

**Pull request merged**